runtime/src/ppl/delite/runtime/profiler/PostProcessorDataStructures.scala

```diff
@@ -0,0 +1,946 @@
+package ppl.delite.runtime
+package profiler
+
+import java.io.{File, PrintWriter}
+import java.sql.{ Connection, DriverManager, ResultSet, SQLException, Statement }
+import ppl.delite.runtime.Config
+import _root_.scala.util.parsing.json.JSON
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable.{ArrayBuffer, HashMap}
+import scala.io.Source
+import tools.nsc.io.Path
+
+object DNodeType extends Enumeration {
+	type DNodeType = Value
+	val WhileLoop, Conditional, MultiLoop, SingleTask, Foreach, EOG, EOP = Value
+}
+
+object TNodeType extends Enumeration {
+	type TNodeType = Value
+	val Kernel, KernelPartition, Sync, TicTocRegion = Value
+}
+
+object Types {
+	type DNodeId = Int
+	type TNodeId = Int
+	type LevelId = Int
+	type DNodeName = String
+	type TNodeName = String
+	type TicTocNodeName = String
+}
+
+object DependencyGraph {
+
+	//implicit conversion to strings
+	implicit def anyToString(obj: Any): String = obj.toString
+
+	val dNodeConditionalSubOpTypes = Array("condOps", "thenOps", "elseOps")
+	val dNodeWhileloopSubOpTypes = Array("condOps", "bodyOps")
+
+	def dependencyGraphNew(degFile: String): DependencyGraph = {
+		val contents = scala.io.Source.fromFile(degFile).mkString
+    	import JSON._
+    	phrase(root)(new lexical.Scanner(contents)) match {
+			case Success(result,_) => buildDepGraphFromParsedJSON(resolveType(result))
+			case f@NoSuccess(_,_) => throw new RuntimeException("Couldn't parse the DEG file:\n" + f.toString)
+    	}
+	}
+
+	private def buildDepGraphFromParsedJSON(json: Any): DependencyGraph = {
+		json match {
+			case degm: Map[Any,Any] => try { 
+				val depGraph = new DependencyGraph
+				val deg = getFieldMap(degm, "DEG")
+				val ops = getFieldList(deg, "ops")
+				depGraph.dNodesNew(ops)
+				return depGraph
+			} catch { case e: Exception => e.printStackTrace; throw e; }
+			case err@_ => mapNotFound(err)
+    	}
+  	}
+
+  	private def getFieldString(map: Map[Any, Any], field:String): String = {
+		map.get(field) match {
+			case Some(field) => field
+			case None => fieldNotFound(field, map)
+		}
+	}
+
+	private def getFieldList(map: Map[Any, Any], field: String): List[Any] = {
+	    map.get(field) match {
+	        case Some(field) => field match {
+	        	case list: List[Any] => list
+	        	case err@_ => listNotFound(err)
+		    }
+
+		    //case None => fieldNotFound(field, map)
+			case None => List[Any]()
+	    }
+  	}
+
+  	private def getFieldMap(map: Map[Any, Any], field: String): Map[Any,Any] = {
+	    map.get(field) match {
+	    	case Some(field) => field match {
+		        case map: Map[Any,Any] => map
+		        case err@_ => mapNotFound(err)
+	      	}
+
+	      	case None => fieldNotFound(field, map)
+	    }
+	}
+
+	private def getSourceContext(op: Map[Any, Any]): SourceContext = {
+		op.get("sourceContext") match {
```

Comment:
the scala (and our) convention for indentation is two spaces... there are several files like this...